build.sbt

```diff
@@ -1324,75 +1324,10 @@ lazy val root: Project = (project in file("."))
-        partestDesc("--srcpath async"),
-        (tasty / Test / Keys.test).result map (_ -> "tasty/test"),
-        (osgiTestFelix / Test / Keys.test).result map (_ -> "osgiTestFelix/test"),
-        (osgiTestEclipse / Test / Keys.test).result map (_ -> "osgiTestEclipse/test"),
-        (library / mimaReportBinaryIssues).result map (_ -> "library/mimaReportBinaryIssues"),
-        (reflect / mimaReportBinaryIssues).result map (_ -> "reflect/mimaReportBinaryIssues"),
-        testJDeps.result map (_ -> "testJDeps"),
-        testJarSize.result map (_ -> "testJarSize"),
-        (bench / Compile / compile).map(_ => ()).result map (_ -> "bench/compile"),
-        Def.task(()).dependsOn( // Run these in parallel:
-          library / Compile / doc,
-          reflect / Compile / doc,
-          compiler / Compile / doc,
-          scalap / Compile / doc,
-        ).result map (_ -> "doc")
-      )).value
-      val log = streams.value.log
-      val failed = results.collect { case (Inc(i), d) => (i, d) }
-      if (failed.nonEmpty) {
-        def showScopedKey(k: Def.ScopedKey[_]): String =
-          Vector(
-            k.scope.project.toOption.map {
-              case p: ProjectRef => p.project
-              case p => p
-            }.map(_ + "/"),
-            k.scope.config.toOption.map(_.name + ":"),
-            k.scope.task.toOption.map(_.label + "::")
-          ).flatten.mkString + k.key
-        val loggedThis, loggedAny = new scala.collection.mutable.HashSet[String]
-        def findRootCauses(i: Incomplete, currentTask: String): Vector[(String, Option[Throwable])] = {
-          val sk = i.node match {
-            case Some(t: Task[_]) =>
-              t.info.attributes.entries.collect { case e if e.key == Keys.taskDefinitionKey => e.value.asInstanceOf[Def.ScopedKey[_]] }
-                .headOption.map(showScopedKey)
-            case _ => None
-          }
-          val task = sk.getOrElse(currentTask)
-          val dup = sk.map(s => !loggedAny.add(s)).getOrElse(false)
-          if(sk.map(s => !loggedThis.add(s)).getOrElse(false)) Vector.empty
-          else i.directCause match {
-            case Some(e) => Vector((task, if(dup) None else Some(e)))
-            case None => i.causes.toVector.flatMap(ch => findRootCauses(ch, task))
-          }
-        }
-        log.error(" ")
-        log.error(s"${failed.size} of ${results.length} test tasks failed:")
-        failed.foreach { case (i, d) =>
-          log.error(s"- $d")
-          loggedThis.clear
-          findRootCauses(i, "<unkown task>").foreach {
-            case (task, Some(ex)) => log.error(s"  - $task failed: $ex")
-            case (task, None)     => log.error(s"  - ($task failed)")
-          }
-        }
-        SavedLogs.showSavedLogsImpl(s => log.error(s))
-        throw new MessageOnlyException("testAll failed due to previous errors")
-      }
-    },
+    testAll  := runTests(unitTests ::: partests ::: osgiTests ::: remainingTests).value,
```

Comment:
I think you should add a comment saying why `testAll` doesn't just invoke `testAll1` and `testAll2`, otherwise that's going to be permanently mysterious to anyone reading this.